internal/cmd/controller/reconciler/clustergroup_controller.go

```diff
@@ -71,6 +72,8 @@ func (r *ClusterGroupReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	})
 	if err != nil {
 		logger.V(1).Error(err, "Reconcile failed final update to cluster group status", "status", group.Status)
+	} else {
+		metrics.CollectClusterGroupMetrics(group)
 	}
```

Comment:
nit: I think this is simpler:
```suggestion
	}
		
	metrics.CollectClusterGroupMetrics(group)
```

Reply:
Oops, this is not functionally equivalent; we may not actually want to collect metrics when we failed to update the status. Please feel free to dismiss my previous comment.